### DISCUSSION TITLE
`--src-dir` is still supported?

### DISCUSSION THREAD

**Discussion by matheusrocha89** [vercel/next.js#44814] [Help]
I saw on `nextjs` documentation that on the `create-next-app` you could use the option `--src-dir` to create the `nextjs` app with the `src` directory but when I try the `CLI` with `--help` this option is not listed. So I put it anyway and nothing happened, no `src` folder was created. 

Should we remove this option from documentation?

[Documentation page](https://nextjs.org/docs/api-reference/create-next-app)

**Comment by icyJoseph** [comment]
I think you may have found a bug... https://github.com/vercel/next.js/blob/canary/packages/create-next-app/templates/index.ts#L126

The CLI is still trying to apply srcDir rules, so something is definitely wrong here.

## Edit

It was actually put back (i it ever was there) in 2 days ago! https://github.com/vercel/next.js/commit/893328905f3507969d2f2ebe8f006901f8bab814 - I guess it'll drop on the next release cycle.

**Reply by matheusrocha89** [reply]
@icyJoseph Do you know the schedule of release cycles of `nextjs`? To be up when this would be fixed?

**Reply by icyJoseph** [reply]
No idea, they do release frequently though.

**Comment by Nefcanto** [comment]
@icyJoseph we are using this command to automate docker image creation:

     RUN npx --yes create-next-app next --use-npm --js --eslint 

And it failed yesterday after they released 13.1.2. We even tried adding the `--src-dir` but made no difference.

How can we fix this?

**Reply by icyJoseph** [reply]
Sorry, I don't really know, what did the failing logs say?

**Reply by Nefcanto** [reply]
@icyJoseph it does not say anything. It just waits for user input and Dockerfile (the `docker build` command) kills the process after some time of inactivity.

**Reply by icyJoseph** [reply]
It finishes to completion on my machine... which I  know is not very helpful to your case... Could you isolate the Docker commands that make it hang and post them here? Including the base image you use and all. Keep private, production stuff out of it.

**Reply by Nefcanto** [reply]
@icyJoseph thank you for helping me. Please follow these steps:

- Create a `Dockerfile`
- Wirte the code below in it
- Use `docker build .` to build it
- You should see that it fails

Dockerfile content:

```
FROM node:lts-bullseye-slim

# RUN apt update \
#     && apt install bc -y \
#     && apt install -y procps \
#     && apt install curl -y \
#     && apt install inotify-tools -y \
#     && apt install jo -y \
#     && apt install jq -y \
#     && apt install git -y \
#     && apt install python3 make g++ -y \
#     && rm -rf /var/cache/apk/* \
#     && npm install --location=global npm -y \
#     && npm install -g npm-check \
#     && cd /usr/local/bin \
#     && curl https://getmic.ro | bash

WORKDIR /

RUN npx --yes create-next-app next --use-npm --js --eslint --src-dir

WORKDIR /next

RUN npm i \
    && npm i --save @date-io/date-fns \
    && npm i --save @date-io/date-fns-jalali \
    && npm i --save @emotion/cache \
    && npm i --save @emotion/react \
    && npm i --save @emotion/server \
    && npm i --save @emotion/styled \
    && npm i --save @fortawesome/fontawesome-svg-core \
    && npm i --save @fortawesome/free-brands-svg-icons \
    && npm i --save @fortawesome/free-regular-svg-icons \
    && npm i --save @fortawesome/free-solid-svg-icons \
    && npm i --save @fortawesome/react-fontawesome@latest \
    && npm i --save @headlessui/react \
    && npm i --save @heroicons/react \
    && npm i --save @mui/base \
    && npm i --save @mui/icons-material \
    && npm i --save @mui/lab \
    && npm i --save @mui/material \
    && npm i --save @next/bundle-analyzer \
    && npm i --save @next/swc-linux-x64-gnu \
    && npm i --save animate.css \
    && npm i --save axios \
    && npm i --save cookieconsent \
    && npm i --save cors \
    && npm i --save date-fns \
    && npm i --save date-fns-jalali \
    && npm i --save formik \
    && npm i --save jsdom \
    && npm i --save marked \
    && npm i --save mixitup \
    && npm i --save nodemailer \
    && npm i --save react-countup \
    && npm i --save react-icons \
    && npm i --save react-youtube \
    && npm i --save recharts \
    && npm i --save sass \
    && npm i --save sharp \
    && npm i --save stylis \
    && npm i --save stylis-plugin-rtl \
    && npm i --save swiper \
    && npm i --save wowjs \
    && npm i --save yup \
    && npm i -D --save-exact prettier \
    && npm i -D @svgr/webpack \
    && npm i -D @tailwindcss/line-clamp \
    && npm i -D @tailwindcss/typography \
    && npm i -D autoprefixer \
    && npm i -D eslint \
    && npm i -D eslint-config-next \
    && npm i -D postcss \
    && npm i -D prettier-plugin-tailwindcss \
    && npm i -D tailwindcss
```

**Reply by icyJoseph** [reply]
I commented out the npm installs, and the docker build went fine without them, are those really necessary (to trigger your issue)? could one of them be the cause of the crash?

**Reply by Nefcanto** [reply]
@icyJoseph  the problem happens exactly at the `create-next-app` line. It's weird that your build went well. Maybe you should update everything first. For me if I delete everything except that line, I still get the error.

**Reply by matheusrocha89** [reply]
@Nefcanto what is the error message that displays?

**Reply by Nefcanto** [reply]
@matheusrocha89 this is an image of the problem:

![Screenshot from 2023-01-19 07-18-53](https://user-images.githubusercontent.com/12782347/213350928-9be2fe9a-3a8e-42bb-b65d-1d3a6d905b68.png)

**Reply by icyJoseph** [reply]
Ah now that's interesting, let me investigate if there's a flag to skip that step. 

I guess your build gets stuck at waiting for an answer to `app-dir`?

**Reply by icyJoseph** [reply]
Looks like create-next-app uses, https://www.npmjs.com/package/ci-info, to calculate whether or not to ask for the app-dir question... now the question is, how do we tell ci-info, that we are in a custom ci, is it even possible? or could we tell Commander to pre-fill some answers?

## EDIT

I am such a noob... there's a `--no-experimental-app` flag...

```js
if (
      !process.argv.includes('--experimental-app') &&
      !process.argv.includes('--no-experimental-app')
    )
```

So your command should be:

```
 RUN npx --yes create-next-app next --use-npm --js --eslint --no-experimental-app --src-dir
```

It became a lit too long I guess...

**Reply by Nefcanto** [reply]
@icyJoseph that worked. Thank you so much. I really appreciate it.
